gateway/src/test/java/io/zeebe/mocks/RequestFutureMock.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.mocks;
+
+import io.zeebe.broker.client.api.ZeebeFuture;
+import io.zeebe.broker.client.api.commands.Topology;
+import io.zeebe.broker.client.api.commands.TopologyRequestStep1;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class RequestFutureMock<E> implements ZeebeFuture<E>, TopologyRequestStep1 {
+
+  private E self;
```

Comment:
Shouldn't we be able to set the self? To return it as result of this future? Currently it is unused